File: backend/app/schemas/form_submission.py
from pydantic import BaseModel
from typing import Optional, Dict, Any
from datetime import datetime
from uuid import UUID


class FormSubmissionBase(BaseModel):
    agency_id: UUID
    assessment_id: Optional[str] = None
    new_assessment_data: Optional[Dict[str, Any]] = None
    pdf_filename: str


class FormSubmissionCreate(FormSubmissionBase):
    pdf_file_data: str  # Base64 encoded


class FormSubmissionOut(FormSubmissionBase):
    id: UUID
    task_id: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: frontend/types/formSubmission.ts
export interface FormSubmission {
    id: string;
    agency_id: string;
    patient_episode_id: string;
    assessment_id?: string;
    new_assessment_data?: Record<string, any>;
    pdf_file_data: string;
    pdf_filename: string;
    task_id?: string;
    created_at: string;
    updated_at: string;
}

export interface FormSubmissionCreate {
    agency_id: string;
    patient_episode_id: string;
    assessment_id?: string;
    new_assessment_data?: Record<string, any>;
    pdf_file_data: string;
    pdf_filename: string;
}

export interface FormSubmissionUpdate {
    task_id?: string;
}

export interface FormSubmissionOption {
    value: string;
    label: string;
    created_at: string;
}
